lib/swagger_client/api_client.rb

```diff
@@ -271,7 +271,7 @@ def sanitize_filename(filename)
     def build_request_url(path)
       # Add leading and trailing slashes to path
       path = "/#{path}".gsub(/\/+/, '/')
-      URI.encode(@config.base_url + path)
+      URI::DEFAULT_PARSER.escape(@config.base_url + path)
```

Comment:
As a suggestion, there could be a wrapper to escape URLs for this gem so next time a change like this happens it affects only one file.